torch/nn/attention/flex_attention.py

```diff
@@ -969,12 +969,13 @@ def _build_seq_idx(offsets, total_length):
     if is_score_mod:
 
         def nt_score_mod(score, b, h, q_idx, kv_idx):
+            b_nested = q_seq_idx[q_idx]
             q_nested = q_idx - q_offsets[q_seq_idx[q_idx]]
             kv_nested = kv_idx - kv_offsets[kv_seq_idx[kv_idx]]
             is_same_sequence = q_seq_idx[q_idx] == kv_seq_idx[kv_idx]
             return torch.where(
```

Comment:
Do we support inplace where yet for this tensor type? Seems like some free performance gains

Reply:
The tensor types here are:
* `q_seq_idx` - standard strided tensor
* `q_idx` - `BatchedTensor` due to vmap usage within flex

Not sure where inplace where would come in, but there is a [custom batching rule](https://github.com/pytorch/pytorch/blob/475656fd9c3871302e164bcd45415f18d68f637b/aten/src/ATen/functorch/BatchRulesIndexing.cpp#L17) for `index.Tensor`.

Reply:
```suggestion
           output = orig_mod_func(score, b_nested, h, q_nested, kv_nested)
           output.where_(is_same_sequence, float("-inf"))
           return output
```

Reply:
Hm I tried this but it's actually not straightforward to use an in-place op here due to vmap:
```python
...
output = orig_mod_func(score, b_nested, h, q_nested, kv_nested)  # type: ignore[call-arg]
output.masked_fill_(q_seq_idx[q_idx] != kv_seq_idx[kv_idx], float("-inf"))
return output
```

```
RuntimeError: vmap: inplace arithmetic(self, *extra_args) is not possible because there exists a Tensor `other` in extra_args that has more elements than `self`. This happened due to `other` being vmapped over but `self` not being vmapped over in a vmap. Please try to use out-of-place operators instead of inplace arithmetic. If said operator is being called inside the PyTorch framework, please file a bug report instead.
```

gonna stick with the current logic for landing